dll/win32/msgina/shutdown.c

```diff
@@ -35,33 +36,52 @@
 #define CY_BITMAP                       33
 
 #define NUMBER_OF_BUTTONS               4
+#define NUMBER_OF_BUTTON_STATES         10
+
+/* For HTHEME and HBITMAP */
+#define DLG_BITMAP_CUSTOMIZABLE         3
+
+#define DLG_BACKGROUND                  0
+#define DLG_FLAG                        1
+#define DLG_BUTTON_BITMAP               2
+#define DLG_STATIC_TITLE                3
+#define DLG_STATIC_BUTTONS              4
+
+/* Static Texts */
+#define STATIC_TURNOFF                  0
+#define STATIC_SHUTDOWN                 1
+#define STATIC_RESTART                  2
+#define STATIC_SLEEP                    3
+#define STATIC_HIBERNATE                4
 
 /* After determining the button as well as its state paint the image strip bitmap using these predefined positions */
 #define BUTTON_SHUTDOWN                 0
-#define BUTTON_SHUTDOWN_PRESSED         (CY_BITMAP + BUTTON_SHUTDOWN)
-#define BUTTON_SHUTDOWN_FOCUSED         (CY_BITMAP + BUTTON_SHUTDOWN_PRESSED)
-#define BUTTON_REBOOT                   (CY_BITMAP + BUTTON_SHUTDOWN_FOCUSED)
-#define BUTTON_REBOOT_PRESSED           (CY_BITMAP + BUTTON_REBOOT)
-#define BUTTON_REBOOT_FOCUSED           (CY_BITMAP + BUTTON_REBOOT_PRESSED)
-#define BUTTON_SLEEP                    (CY_BITMAP + BUTTON_REBOOT_FOCUSED)
-#define BUTTON_SLEEP_PRESSED            (CY_BITMAP + BUTTON_SLEEP)
-#define BUTTON_SLEEP_FOCUSED            (CY_BITMAP + BUTTON_SLEEP_PRESSED)
-#define BUTTON_SLEEP_DISABLED           (CY_BITMAP + BUTTON_SLEEP_FOCUSED)
+#define BUTTON_SHUTDOWN_PRESSED         cy
+#define BUTTON_SHUTDOWN_FOCUSED         (cy * 2)
+#define BUTTON_REBOOT                   (cy * 3)
+#define BUTTON_REBOOT_PRESSED           (cy * 4)
+#define BUTTON_REBOOT_FOCUSED           (cy * 5)
+#define BUTTON_SLEEP                    (cy * 6)
+#define BUTTON_SLEEP_PRESSED            (cy * 7)
+#define BUTTON_SLEEP_FOCUSED            (cy * 8)
+#define BUTTON_SLEEP_DISABLED           (cy * 9)
 
 typedef struct _SHUTDOWN_DLG_CONTEXT
 {
     PGINA_CONTEXT pgContext;
-    HBITMAP hBitmap;
-    HBITMAP hImageStrip;
-    DWORD ShutdownOptions;
-    HBRUSH hBrush;
-    HFONT hfFont;
     BOOL bCloseDlg;
-    BOOL bIsSleepButtonReplaced;
-    BOOL bReasonUI;
     BOOL bFriendlyUI;
     BOOL bIsButtonHot[NUMBER_OF_BUTTONS];
-    BOOL bIsDialogModal;
+    BOOL bIsSleepButtonReplaced;
+    BOOL bReasonUI;
+    BOOL bTimer;
+    HBITMAP hBitmap[3];
+    DWORD ShutdownDialogId;
+    DWORD ShutdownOptions;
+    HFONT hfFont;
+    HTHEME hTheme[5];
+    UINT_PTR iTimer;
+    WCHAR StaticTexts[NUMBER_OF_BUTTONS + 1][128];
```

Comment:
`NUMBER_OF_BUTTONS` _**plus**_ one? why that "+1" ?

Reply:
@HBelusca Static texts for buttons + Static text for heading.
I will get rid of the magic values.